shards/core/ref_output_pool.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef EF63CEF1_3538_4C10_B6F5_75FE1E4D2F44
+#define EF63CEF1_3538_4C10_B6F5_75FE1E4D2F44
+
+#include <list>
+#include <cassert>
+
+namespace shards {
+
+template <typename T> struct RefOutputPoolItemTraits {
+  T newItem() { assert("not implemented"); }
+  void release(T &) { assert("not implemented"); }
+  size_t getRefCount(T &v) { assert("not implemented"); }
+};
+
+// Keeps a pool of output objects
+// the objects are assumed to be self ref counted, so destruction is not handled by the pool
+template <typename T> struct RefOutputPool {
```

Comment:
We probably wanna add a dedicated test in test-runtime? Also maybe some `shassert` for runtime to avoid leaking somehow?

Reply:
Added